lib/db.js

```diff
@@ -123,6 +144,10 @@ class LimitDBRedis extends EventEmitter {
     return Math.ceil((now + msToCompletion) / 1000);
   }
 
+  /**
+   * @param {bucketConfigParams} params
+   * @returns {void} nothing if valid, ValidationError if invalid
+   */
   validateParams(params) {
```

Comment:
Should we validate the contents of the optional `params.configOverride` attribute? For example, if it is passed, it should not be an empty object (it should at least have a size or a refill rate).

Reply:
I think this relates to the above ☝️ comment.  If we're using an either/or logic to *either* use the overrides, *or* the existing buckets, then we have to be more stringent on validation.  Since merging the properties will pretty much always fall back to a sane default, validation can be looser.  Here, we'll need to decide to either:

- Error when `configOverride` is passed, but invalid
- Do something like warn/emit even when `configOverride is passed, but invalid, and use defaults

Preferences?

Reply:
Because one of the strange things, is that when a `size` is not explicitly provided, it is automatically set to the `per_interval`.  So these two configs would both yield a bucket size of 10:

- `per_day`: 10
- `per_second`: 10

Although those seem to be drastically different configs.  Do we want to replicate that as well?

Reply:
My preference would be to "Error when `configOverride` is passed, but invalid". We should already have the safe fallback error handling in suspicious IP throttling (skipping when limitr throws).

> Do we want to replicate that as well?

yes, I think so. To me, this is working as expected because it is already like that when passing static bucket configs.

Reply:
Okie dokie... when rates are passed in, they're used as max size.